ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/SqlToJavaVisitor.java

```diff
@@ -478,6 +478,42 @@ private String visitStringComparisonExpression(final ComparisonExpression.Type t
       }
     }
 
+    private String visitArrayComparisonExpression(final ComparisonExpression.Type type) {
+      switch (type) {
+        case EQUAL:
+          return "java.util.Arrays.deepEquals(%1$s.toArray(), %2$s.toArray())";
```

Comment:
is there any reason to convert to an array? I believe we could just use `List#equals`

Reply:
+1